pkgs/by-name/fi/filebot/package.nix

```diff
@@ -63,9 +70,25 @@ stdenv.mkDerivation (finalAttrs: {
       --replace '$FILEBOT_HOME/data/.license' '$APP_DATA/.license' \
       --replace '-jar "$FILEBOT_HOME/jar/filebot.jar"' '-Dcom.googlecode.lanterna.terminal.UnixTerminal.sttyCommand=${coreutils}/bin/stty -jar "$FILEBOT_HOME/jar/filebot.jar"'
     wrapProgram $out/opt/filebot.sh \
-      --prefix PATH : ${lib.makeBinPath [ openjdk17 ]}
+      --prefix PATH : ${
+        lib.makeBinPath [
+          chromaprint
```

Comment:
Can you add a comment why these dependencies are needed?

Reply:
And also why they weren't put into the path before.

Reply:
> Can you add a comment why these dependencies are needed?

These are dependencies the filebot author said are needed for all the features to work and be available. 

I'm kind of new to nix lang so maybe I got things in the wrong places. I mostly have worked with flakes so far and I thought buildInputs is where runtime dependencies go, while nativeBuildInputs is where build and dev dependencies go.

Reply:
This here is only for adding executables that the program needs at runtime to the program's `$PATH`. If it requires a dependency for anything other than executing one of its executables in `bin/`, this is the wrong place.

Reply:
Ah, thank you. Ok, I'll remove the libs and leave the executable dependencies here. Thanks.

Reply:
I clicked the wrong button and started a review :( Sorry. Not sure how to undo that.

Reply:
Please ensure the executables are actually required by the program and that all required executables are present by running it with `PATH= ./result/bin/program`.

The review thing doesn't matter, it just bundles your replies for bulk submission.

Reply:
The review thing doesn't matter, it just bundles your replies for bulk submission. For future reference: You can simply delete the review comments again before submission (ideally copy the text first).

Reply:
So, I ran `nix build --impure .#filebot` and now have a './result/bin/filebot', but running that command you gave gives me:

```
PATH= ./result/bin/filebot
/nix/store/iawl2rnqvav1ppfh5as59gybya4awi6v-filebot-5.1.5/opt/.filebot.sh-wrapped: line 16: dirname: command not found
/nix/store/iawl2rnqvav1ppfh5as59gybya4awi6v-filebot-5.1.5/opt/.filebot.sh-wrapped: line 33: uname: command not found
/nix/store/iawl2rnqvav1ppfh5as59gybya4awi6v-filebot-5.1.5/opt/.filebot.sh-wrapped: line 33: uname: command not found
Error: Unable to access jarfile /home/david/repos/nixpkgs/pkgs/by-name/fi/filebot/jar/filebot.jar
```

Considering I didn't add anything using dirname or uname that I know of, I must be executing it wrong. How do I do this? And thanks for all the guidance.

Reply:
> > Can you add a comment why these dependencies are needed?
> 
> These are dependencies the filebot author said are needed for all the features to work and be available.
> 
> I'm kind of new to nix lang so maybe I got things in the wrong places. I mostly have worked with flakes so far and I thought buildInputs is where runtime dependencies go, while nativeBuildInputs is where build and dev dependencies go.
> 
> Why they weren't in the path before I don't know. I might also have this wrong here because I can get zenity to spawn windows when manually in the nix shell but filebot still can't. I thought I was configuring it wrong though. Advice?

Further info on dependencies: p7zip and unrar are used by filebot to unarchive various distributions of media that may be archived, even with nested archives.

Chromaprint provides fpcalc tool, which is used to identify and catalogue music and music videos.

Reply:
Ah, you'd need to add basic utilities such as coreutils to the path of course. My intention with unsetting path was to ensure none of its runtime dependencies were missed by them already existing in your system-wide path.

Coreutils can be assumed to be present though.